File: tmdb-core/src/main/java/dev/reuss/tmdb/value/media/MediaType.java
package dev.reuss.tmdb.value.media;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * TMDB media type discriminator used in mixed responses.
 */
public enum MediaType {
    MOVIE("movie"),
    TV("tv"),
    PERSON("person"),
    UNKNOWN("unknown");

    private final String value;

    MediaType(String value) {
        this.value = value;
    }

    /**
     * Returns the TMDB media type value.
     *
     * @return media type value
     */
    @JsonValue
    public String value() {
        return value;
    }

    /**
     * Parses a TMDB media type value.
     *
     * @param value raw media type value
     * @return matching media type, or {@link #UNKNOWN} if the value is absent or unknown
     */
    @JsonCreator
    public static MediaType fromValue(String value) {
        if (value == null) {
            return UNKNOWN;
        }

        for (MediaType mediaType : values()) {
            if (mediaType.value.equals(value)) {
                return mediaType;
            }
        }

        return UNKNOWN;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/media/MediaType.kt
package dev.reuss.tmdb.value.media

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

/**
 * TMDB media discriminator used by mixed search and trending responses.
 */
enum class MediaType(
    @get:JsonValue
    val value: String,
) {
    MOVIE("movie"),
    TV("tv"),
    PERSON("person"),
    UNKNOWN("unknown"),
    ;

    companion object {
        /**
         * Maps a wire value to a known media type without throwing.
         *
         * @return matching media type, or [UNKNOWN] if the value is absent or unknown
         */
        @JvmStatic
        @JsonCreator
        fun fromValue(value: String?): MediaType = entries.firstOrNull { it.value == value } ?: UNKNOWN
    }
}
